### PR TITLE
Vulkan: sort free texture memory regions by size

### PR DESCRIPTION
**Pull request by thatcosmonaut**
Instead of iterating over every texture memory page and free region when looking for available texture memory, we now maintain a list of free regions at the allocator level, sorted by size.

### CODE REVIEW

File: src/FNA3D_Driver_Vulkan.c [hunk 1]
```diff
@@ -947,31 +947,44 @@ typedef struct BufferMemoryWrapper
...
+	 * from largest to smallest for efficient texture allocation
+	 */
+	VulkanFreeTextureRegion **sortedFreeRegions;
+	uint32_t sortedFreeRegionCount;
+	uint32_t sortedFreeRegionCapacity;
+} VulkanTextureSubAllocator;
+
 /* Manages memory for a single region of allocated texture memory */
-typedef struct VulkanTextureBuffer
+struct VulkanTextureBuffer
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_Vulkan.c:978]: Should this still be a typedef?

**Review by flibitijibibo — COMMENTED** [review]
Not much to see here, will check performance first thing in the morning.

File: src/FNA3D_Driver_Vulkan.c (hunk 1 above)
Comment (thatcosmonaut) [reply src/FNA3D_Driver_Vulkan.c:978]: It has to be forward-declared because of mutual reference.

Comment (flibitijibibo) [reply src/FNA3D_Driver_Vulkan.c:978]: Woops, totally missed that, my mistake!

**Pull request merged**